src/main/java/com/blamejared/compat/tcomplement/highoven/HighOven.java

```diff
@@ -0,0 +1,305 @@
+			return LogHelper.getStackDescription(output);
+		}
+	}
+
+	private static class RemoveHeat extends BaseAction {
+		private ILiquidStack input;
+		private ILiquidStack output;
+
+		public RemoveHeat(ILiquidStack input, ILiquidStack output) {
+			super("HighOven.HeatRecipe");
+			this.input = input;
+			this.output = output;
+		}
+
+		@Override
+		public void apply() {
+			FluidStack output = InputHelper.toFluid(this.output);
+			if (this.input == null) {
+				REMOVED_HEAT_RECIPES.put(output, null);
+			} else if (REMOVED_HEAT_RECIPES.containsKey(output)) {
+				Set<FluidStack> inputs = REMOVED_HEAT_RECIPES.get(output);
+				if (inputs != null)
+					inputs.add(InputHelper.toFluid(input));
+			} else {
+				Set<FluidStack> inputs = new HashSet<>();
+				inputs.add(InputHelper.toFluid(input));
+				REMOVED_HEAT_RECIPES.put(output, inputs);
+			}
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		public String getRecipeInfo() {
+			return LogHelper.getStackDescription(output)
+					+ ((input == null) ? "" : (" from " + LogHelper.getStackDescription(input)));
+		}
+
+	}
+
+	@ZenMethod
+	public static void removeMixRecipe(ILiquidStack output, @Optional ILiquidStack input) {
+		init();
+		CraftTweakerAPI.apply(new RemoveMix(output, input));
+	}
+
+	@ZenMethod
+	public static MixRecipeBuilder newMixRecipe(ILiquidStack output, ILiquidStack input, int temp) {
+		init();
+		return new MixRecipeBuilder(output, input, temp);
+	}
+	
+	@ZenMethod
+	public static MixRecipeManager manageMixRecipe(ILiquidStack output, ILiquidStack input) {
+		init();
+		return new MixRecipeManager(output, input);
+	}
+
+	private static class RemoveMix extends BaseAction {
+		private ILiquidStack output;
+		private ILiquidStack input;
+
+		public RemoveMix(ILiquidStack output, ILiquidStack input) {
+			super("HighOven.MixRecipe");
+			this.output = output;
+			this.input = input;
+		}
+
+		@Override
+		public void apply() {
+			FluidStack output = InputHelper.toFluid(this.output);
+			if (this.input == null) {
+				REMOVED_MIX_RECIPES.put(output, null);
+			} else if (REMOVED_MIX_RECIPES.containsKey(output)) {
+				Set<FluidStack> inputs = REMOVED_MIX_RECIPES.get(output);
+				if (inputs != null)
+					inputs.add(InputHelper.toFluid(input));
+			} else {
+				Set<FluidStack> inputs = new HashSet<>();
+				inputs.add(InputHelper.toFluid(input));
+				REMOVED_MIX_RECIPES.put(output, inputs);
+			}
+		}
+
+		@Override
+		public String describe() {
+			return String.format("Removing %s Recipe(s) for %s", this.name, this.getRecipeInfo());
+		}
+
+		@Override
+		protected String getRecipeInfo() {
+			return LogHelper.getStackDescription(output)
+					+ ((input == null) ? "" : (" from " + LogHelper.getStackDescription(input)));
+		}
+	}
+
+	@SubscribeEvent
+	public void onTinkerRegister(TCompRegisterEvent.HighOvenFuelRegisterEvent event) {
```

Comment:
I would suggest giving these different names just to make it easier on stacktraces